RFC/src/RFC-0120_Consensus.md

```diff
@@ -58,26 +58,60 @@ The aim of this Request for Comment (RFC) is to describe the fields that a block
 
 ## Description
 
-### Blocks
-
-Every [block] MUST conform to the following:
+Blockchain consensus is a set of rules that a majority of nodes agree on, that determines the state of the blockchain. 
 
-* Have a _single_ valid coinbase [UTXO] and kernel
-* have a _single_ valid blockheader;
-* every [UTXO] has a valid [range proof].
+This RFC details the consensus rules for the Tari network. 
 
-If a [block] does not confirm to the above, the block should be rejected as invalid and the peer from which it was received marked as a malicious peer.
+### Blocks
+[Blocks]: #blocks "Block consensus"
+
+Every [block] MUST:
+
+* have _exactly one_ valid [block header], as per the [Block Headers] section
+* have _exactly one_ [coinbase] transaction
+* have a total [transaction weight] less than the consensus maximum
+* be able to calculate matching Merkle roots ([kernel_mr], [output_mr], [witness_mr] and [input_mr]) 
+* each [transaction input] MUST: 
+  * spend an existing valid [UTXO] 
+  * have a maturity greater than the current block height
+  * be in a canonical order (see [Transaction ordering])
+* each [transaction output] MUST:
+  * have a unique hash (`features || commitment || script`)
+  * be in a canonical order (see [Transaction ordering])
+  * have a valid [range proof]
+  * have a valid [metadata signature]
+  * have a valid script offset (\\gamma), as per [RFC-0201_TariScript](./RFC-0201_TariScript.md).
+* each [transaction kernel] MUST have a valid kernel signature
+* the transaction commitments and kernels MUST balance, as follows:
+
+  $$
+  \begin{align}
+  \sum_i\mathrm{Cin_{i}} - \sum_j\mathrm{Cout_{j}} + \text{fees}H \stackrel{?}{=} \sum_k\mathrm{K_k} + \text{offset}
+  \\ \text{for each input} j,
+  \\ \text{each output}, i,
+  \\ \text{and each kernel excess}, k
+  \\ \textit{offset }\text{is the total kernel offset}
+  \end{align}
+  \tag{1}
+  $$
+
+
+If a [block] does not conform to the above, the block SHOULD be discarded and MAY ban the peer that sent it.
 
 #### Coinbase
+[coinbase]: #coinbase "Coinbase consensus"
 
-Every coinbase transaction contained in a block MUST conform to the following:
+A coinbase transaction contained in a block MUST:
 
-* Be only one UTXO
-* Have only one kernel
-* Contain the exact specified emission amount
-* Contain the exact specified lock-height.
+* be the only transaction in the block with the coinbase flag
+* consist of exactly one output and one kernel (no input)
+* have a valid kernel signature
+* have a value exactly equal to the emission at the block height it was minted (see [emission schedule]) 
+  plus the total transaction fees within the block
+* have a lock-height as per consensus
```

Comment:
I know we have not formally decided on a number, but we should try to update this as we have numbers so someone only has to read these and does not have to search the code base for the actual value.

Reply:
yeah agreed, didn't want to publish numbers before they were final, so we should add that to tasks for after mainnet.